src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java

```diff
@@ -1,16 +1,28 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import java.util.NoSuchElementException;
+import mate.academy.hibernate.relations.dao.MovieDao;
+import mate.academy.hibernate.relations.dao.impl.MovieDaoImpl;
 import mate.academy.hibernate.relations.model.Movie;
 import mate.academy.hibernate.relations.service.MovieService;
+import org.hibernate.SessionFactory;
 
 public class MovieServiceImpl implements MovieService {
+    private MovieDao movieDao;
+
+    public MovieServiceImpl(SessionFactory sessionFactory) {
```

Comment:
It's recommended to use dependency injection for the DAO layer instead of instantiating `MovieDaoImpl` directly in the constructor. This allows for better decoupling and easier testing. Consider adding a constructor that accepts a `MovieDao` as a parameter or using a dependency injection framework.

Reply:
It's recommended to use dependency injection for the DAO layer instead of instantiating MovieDaoImpl directly in the constructor. This allows for better decoupling and easier testing. Consider adding a constructor that accepts a MovieDao as a parameter or using a dependency injection framework.